pkg/cvo/status.go

```diff
@@ -413,6 +420,41 @@ func updateClusterVersionStatus(cvStatus *configv1.ClusterVersionStatus, status
 	}
 }
 
+// filterErrorForFailingCondition filters out update errors based on the given
+// updateEffect. If the err matches the given update effect, nil is
+// returned. If the err has the reason MultipleErrors, its immediate nested
+// errors are filtered out and the error is recreated. Otherwise, the original
+// err is returned.
+func filterErrorForFailingCondition(err error, updateEffect payload.UpdateEffectType) error {
+	if uErr, ok := err.(*payload.UpdateError); ok {
+		if uErr == nil || uErr.UpdateEffect == payload.UpdateEffectNone {
```

Comment:
I don't think we need this `uErr.UpdateEffect == payload.UpdateEffectNone` check.  I suspect it's there because you want to avoid touching the error if all the children are `UpdateEffectNone`.  But I don't have a problem rebuilding errors like that with a fresh `newMultipleError(filtered)`, to avoid having to make assumptions about how the incoming error was constructed.

Reply:
I'm not sure I follow, it looks legit to me? If the top level input error is `UpdateEffectNone`  why not returning right away? Are we worred that a top-level `UpdateEffectNone` error may contain non-`UpdateEffectNone` children?

Reply:
The check functions to filter any `err` that declares its impact as `UpdateEffectNone`. A single non-nested `UpdateEffectNone` or any aggregate `UpdateEffectNone` errors (that may include any number of nested errors). It functions to basically ignore a root error that knows its overall impact, and we know by the impact that the error is "_is not worthy"_ to be shown in the `Failing` condition. For example, the test case `single UpdateEffectNone error is not propagated to the condition` tests this case.

The `convertErrorToProgressing` sets the variable `skipFailure` to `False` in many various cases. For example, a history is empty -> `skipFailure == False`. And then the filtering notices that the error is not needed to be shown in the `Failing` condition.

https://github.com/openshift/cluster-version-operator/blob/90e49c6875edae3ad8ae6f2cb54ff6a3749b0200/pkg/cvo/status.go#L482-L485

> But I don't have a problem rebuilding errors like that with a fresh newMultipleError(filtered), to avoid having to make assumptions about how the incoming error was constructed.

I don't think I follow. Could you please elaborate?

Reply:
The mentioned code got changed, and there are no follow-ups. I am marking the discussion as resolved.